### PR TITLE
Improvements to tooling

### LINKED ISSUE #1: Improvements to tooling
I'm proposing the following improvements to tooling:

- Vite setup for remix
    - Default way of creating remix apps as of Mar 2024
    - Faster builds
    - Faster HMR
    - Can tap into larger vite ecosystem of plugins

- [Vite imagetools](https://www.npmjs.com/package/vite-imagetools) plugin to pre-process images at build time and serve them in a more efficient format (webp).

- Typescript
    - Just because every remix project has it these days?
    - Adds some noise to the code (but hopefully you'll soon tune it out)

- [Biome](https://biomejs.dev/) as linter and formatter (replacing prettier + eslint)
    - Way, way faster
    - Will need new contributors to install a vscode plugin for biome

- Bun as a package manager
    - Faster installs
    - But needs contributors to have `bun` installed

- Tailwind -> 4.alpha
    - Faster
    - CSS based config file (as opposed to the js based one)

- MDX based docs (replaces `marked`)
    - More extensible format
    - Can add custom react components in markdown if ever needed

- [Shiki](https://shiki.style/) for syntax highlighting (replaces `highlight.js`)
    - Way faster
    - Zero runtime - can generate styles at build time (low overhead on client side)
    - Used by the Vue.js documentation team

### PR DESCRIPTION
**Pull request by chrzrdx**
Closes https://github.com/sorokya/reoserv.net/issues/1

**Tooling improvements**

- [x] [Bun](https://bun.sh/) as a package manager
- [x] [Biome](https://biomejs.dev/) as formatter for js/ts/jsx/tsx/css/html/json files
- [x] [Prettier 3](https://prettier.io/) as formatter only for markdown files
- [x] Remove eslint in favour of the biome linter
- [x] Vite setup for remix
- [x] Tailwind -> 4.alpha
- [x] Update all dependencies to their latest versions
- [x] [Shiki](https://shiki.style/) for syntax highlighting (replaces highlight.js)- [ ] Typescript
- [x] [Vite imagetools](https://www.npmjs.com/package/vite-imagetools) plugin to pre-process images at build time and serve them in a more efficient format (webp).
- [ ] MDX based docs (replaces marked)
- [ ] Typescript

**Other goodies**

- [x] Add a README
- [x] Add LICENSE, CONTRIBUTING, CODE_OF_CONDUCT guides
- [x] Refactor server utilities
- [x] Misc. bug fixes
- [x] Prefetch links when the user hovers on them (for faster navigations)
- [x] Load analytics script only in production environments

### CODE REVIEW

File: LICENSE.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+MIT License
+
+Copyright (c) [2024] [sorokya](https://github.com/sorokya)
```
Comment (sorokya) [inline LICENSE.md]: Can use my name here "Richard Leek"

File: README.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Reoserv.net documentation home
+
+This website hosts the documentation for the [reoserv][reoserv] project
+(authored and maintained by [sorokya][sorokya])
```
Comment (sorokya) [inline README.md]: Can use my name here "Richard Leek"

File: README.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+Verify that both are setup on your system and meet our version requirements by
+running:
+
+```sh
+bun --version    # eg: 1.0.36
+node --version   # eg: v20.4.0
+```
+
+Also, install the following VS Code extensions:
```
Comment (sorokya) [inline README.md]: I'd denote this as optional and maybe just link to the docs for the tools rather than the vscode marketplace

**Review by sorokya — CHANGES_REQUESTED** [review]
Just a couple changes requested in the markdown files. But you're a legend. I LOVE these changes. Thank you so much 😊 

I'll test these out Monday and hopefully can get merged

**Comment by chrzrdx** [comment]
> Just a couple changes requested in the markdown files. But you're a legend. I LOVE these changes. Thank you so much 😊
> 
> I'll test these out Monday and hopefully can get merged

Submitted those changes you requested for

**Review by sorokya — APPROVED** [review]
Some default styling issues with the new tailwind but I suspect this will be fixed with #8

**Pull request merged**